### PR TITLE
Fix cart summary display when selecting another carrier

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | In checkout, when selecting another carrier cart summary was breaking |
| Type? | bug fix |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-777 |
| How to test? | select another carrier in checkout |

### CODE REVIEW

**Comment by vincentbz** [comment]
Tested and ok, thanks !

File: themes/classic/_dev/js/checkout.js [hunk 1]
```diff
@@ -17,11 +17,14 @@ function setupMyCheckoutScripts() {
...
   });
 
-  let url = `${$('.js-terms a').attr('href')}?content_only=1`;
-
-  $.get(url, (content) => {
-    $('#modal').find('.modal-content').html($(content).find('.page-cms').contents());
-  });
+  var url = $('.js-terms a').attr('href');
+  if (url) {
+    // TODO: Handle request if no pretty URL
```
Comment (mickaelandrieu) [inline themes/classic/_dev/js/checkout.js]: what happens if no pretty url ?

**Comment by mickaelandrieu** [comment]
Thank you @julienbourdeau, do we need to update starter theme too ?

File: themes/classic/_dev/js/checkout.js (hunk 1 above)
Comment (julienbourdeau) [inline themes/classic/_dev/js/checkout.js]: it breaks :trollface:

File: src/Adapter/Cart/CartPresenter.php [hunk 2]
```diff
@@ -230,12 +230,16 @@ public function present($cart)
...
-        $subtotals['discounts'] = array(
-            'type' => 'discount',
-            'label' => $this->translator->trans('Discount', [], 'Cart'),
-            'amount' => $total_discount,
-            'value' => $this->priceFormatter->format($total_discount),
-        );
+        if ($total_discount) {
+            $subtotals['discounts'] = array(
+                'type' => 'discount',
+                'label' => $this->translator->trans('Discount', [], 'Cart'),
```
Comment (Quetzacoalt91) [inline src/Adapter/Cart/CartPresenter.php]:
```php
$this->translator->trans('Discount', array(), 'Cart'),
```

**Comment by Shudrum** [comment]
Thank you @mickaelandrieu, do not forget to do this if needed.

**Pull request merged**